### DISCUSSION TITLE
How to create new resource from a file via CKAN API with cURL?

### DISCUSSION THREAD

**Discussion by btran2195** [ckan/ckan#6141] [Q&A]
The following cURL is from: https://tech.datopian.com/ckan/play-around.html#create-organizations-datasets-and-resources and lets you add a resource to a datset from a url.

curl -X POST http://ckan:5000/api/3/action/resource_create -H "Authorization: 9c04a69d-79f4-4b4b-b4e1-f2ac31ed961c" -d '{ "package_id": "my-first-dataset", "url": "https://raw.githubusercontent.com/frictionlessdata/test-data/master/files/csv/100kb.csv", "description": "This is the best resource ever!" , "name": "brand-new-resource" }'

Was wondering how you could add a resource from a file.

**Comment by jqnatividad** [comment]
Instead of curl, I would use `ckanapi` instead as it affords a more robust CLI interface.

https://github.com/ckan/ckanapi#-create-a-resource-with-a-file-attached

`
$ ckanapi action resource_create package_id=my-dataset-with-files upload@/path/to/file/to/upload.csv
`

**Reply by btran2195** [reply]
To create a dataset using the CLI interface would it be something like this:

ckanapi action package_create -r 'remote instance' -a 'API Key' { "name": "dataset name", ... }

Not sure how to add the parameters in the CLI interface.

**Reply by anuveyatsu** [reply]
@btran2195 I haven't tried myself but I think you can pass a json file with package metadata:

```bash
ckanapi action package_create -I dataset.json
```
